### ISSUE TITLE
[Request] Native win64 Build Instructions

### ISSUE THREAD

**Issue by LordNyriox** [MultiMC/MultiMC5#2394]
System Information
-----------------------------
MultiMC version:  **0.6.x**

Operating System:  **Windows 10 x64**

Summary of the issue or suggestion:
----------------------------------------------
Please add instructions for building MultiMC as a **native 64-bit** program, to `BUILD.md`. 

What should happen:
------------------------------
Currently, both Linux and macOS instructions are provided for building 64-bit MultiMC binaries.  However, for the Windows OS, the *only* build-instructions that are provided, are for 32-bit MultiMC binaries. 

This is a problem, both due to cross-platform consistency, and due to effective opacity.  Not all users who desire to create their own builds, have the necessary knowledge or experience to adapt 32-bit instructions, into their 64-bit equivalents. 

And given that there are security and efficiency benefits (not just performance ones) to using native 64-bit programs, there are likely any number of users who happen to be in this position. 

Though an argument can be made in favor of going whole-hog and providing 64-bit Windows builds directly, I believe the win64 transition could be more gradual. 

The first step, as always, should be to document what the steps are, when it comes to building MultiMC for the 64-bit Windows platform. 

After that is done, a new issue can be posted regarding the possibility of official win64 native builds.  

Places I have looked:
------------------------------------------------
1. Official Builds: [<https://multimc.org/#Download>] 

2. `develop` Branch: [<https://github.com/MultiMC/MultiMC5/blob/develop/BUILD.md>] 

3. `stable` Branch: [<https://github.com/MultiMC/MultiMC5/blob/stable/BUILD.md>] 

4. MultiMC Wiki: [<https://github.com/MultiMC/MultiMC5/wiki>]

Suspected cause:
---------------------------
It is common to discount the benefits of providing 64-bit builds of many smaller programs on Windows. 

While it is true that some programs will use more memory when built in 64-bit, it is equally true that most present-day systems have more than enough memory to absorb this cost. 

While it is likewise true that smaller programs will see less significant performance gains by the change to 64-bit, it is equally true that even small improvements can be a good thing. 

There are also system security features that only 64-bit programs can benefit from, but again, that aspect is often discounted as insignificant.

**Comment by peterix** [comment]
TL;DR: The definitive answer is no.

-----

The 64bit linux build exists because there are 32bit-only linux distributions.
The 32bit linux build exists because there are 64bit-only linux distributions.
The 64bit macos build exists all supported versions of macos are currently 64bit only.
The 32bit Windows build is sufficient for all Windows users.

The only reasons why I would consider this are the following:
* MultiMC requiring more than 2GB of memory - that's not going to happen. If it happens, it's a bug, and I'd rather fix the bug.
* Microsoft releasing a 64bit-only version of Windows aimed at desktops. Windows Server does not count, and you can install 32bit compatibility there anyway.

On top of that, there is a cost associated with setting up CI for it and maintaining it. I am not willing to pay for it. Not with money (extra Windows licenses, extra storage for VMs, S3 hosting costs), and not with my time. It's simply not worth any of ther marginal benefits it may bring (and I'm not convinced it can help anyone on desktop Windows anyway, the platform is a complete trainwreck security-wise).

**Comment by LordNyriox** [comment]
@peterix:  I only requested build **instructions**, not actual builds, or any kind of actual implementation. 

Or am I somehow mistaken in my belief that building win64 binaries is not as simple as substituting "64" for every "32" within the build instructions?

**Comment by LordNyriox** [comment]
@peterix:  I only requested build **instructions**, not actual builds, or any kind of actual implementation.  

I wish to be able to manually create win64 builds for my own use, but I am insufficiently familiar with code-building routines to know the steps, much less how to adapt them. 

Or am I somehow mistaken in my belief that building win64 binaries **is not** as simple as substituting "64" for every "32" within the build instructions?

**Comment by peterix** [comment]
There is no point to doing it... so why would I? And why would you?

**Comment by LordNyriox** [comment]
@peterix:  Diminishing returns, do not inherently mean no returns at all. 

64-bit builds generally provide marginal performance improvements (on compatible systems), as compared to equivalent 32-bit builds—even in smaller programs such as MultiMC.  

Given how resource-hungry Minecraft is, I would like to be able to squeeze that tiny bit of extra efficiency out of my falling-apart old laptop.  

(And before you ask, I am not in such a financial position that I can reasonably replace my rapidly aging machine.)

**Comment by peterix** [comment]
Loss. Pure loss.

**Comment by LordNyriox** [comment]
@peterix:  And is there an effective alternative?  

I have already optimized my system (and Java) configuration as much as I can figure out how to do, yet all the trickery (and Optifine) in the world is not keeping my framerate above the 40's.

**Comment by peterix** [comment]
When the game is running (and you can totally run 64bit java from 32bit multimc, if you so please), MultiMC has no perceptible effect on it. They are two distinct processes and MultiMC is not doing much.

So your problem is that the game is poorly optimized, and that your hardware is essentially a potato.

**Comment by peterix** [comment]
In any case, I have no idea what sort of hardware we are talking about ... soooo.

Please attach the full dxdiag report so we can figure out what is wrong. You can do it like this:

* Press the `[Windows]+[R]` keys on your keyboard to bring up the Run command.
* In the `Run` command box, type in `dxdiag` and press `<Enter>`. The DirectX Diagnostic Tool pop-up window should appear.
* Click on `Save all Information.`
* Save the information to your Desktop, keep the file name as is.
* Attach the file here.

Here is an example of what the DirectX Diagnostic Tool will look like:

![dxdiag example](https://user-images.githubusercontent.com/203326/34328124-4489aa8e-e8d6-11e7-9984-0f5e8561564b.png)
